src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -907,15 +864,14 @@ VideoDecoder::AVFrameStream VideoDecoder::decodeAVFrame(
           getFFMPEGErrorStringFromErrorCode(ffmpegStatus));
     }
 
-    if (activeStreamIndices_.count(packet->stream_index) == 0) {
-      // This packet is not for any of the active streams.
+    if (packet->stream_index != activeStreamIndex_) {
```

Comment:
BTW, this is the "demux" part now. And I think we should be calling `av_read_frame` within its own `while` loop, for as long as the received packet isn't of the target stream.

Reply:
Yeah, what we're currently doing is inefficient, because if the packet is not the right stream, we first have to make a call to `avcodec_receive_frame()` that we know will fail. We're still doing the correct thing (I think), it's just inefficient.